huaweicloud/services/dew/data_source_huaweicloud_kps_keypairs.go

```diff
@@ -0,0 +1,167 @@
+package dew
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/huaweicloud/huaweicloud-sdk-go-v3/services/kps/v3/model"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/helper/hashcode"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func DataSourceKeypairs() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceKeypairsRead,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
```

Comment:
`Computed` and `ForceNew` are unnecessary.

Reply:
fixed